_posts/2018-XX-XX-running-kafka-on-dedicated-nodes.md

```diff
@@ -0,0 +1,341 @@
+```
+
+With the Cluster Operator running, Kafka can be deployed using following resource:
+
+```
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    replicas: 3
+    readinessProbe:
+      initialDelaySeconds: 15
+      timeoutSeconds: 5
+    livenessProbe:
+      initialDelaySeconds: 15
+      timeoutSeconds: 5
+    config:
+      offsets.topic.replication.factor: 3
+      transaction.state.log.replication.factor: 3
+      transaction.state.log.min.isr: 2
+    tolerations:
+      - key: "dedicated"
+        operator: "Equal"
+        value: "Kafka"
+        effect: "NoSchedule"
+    storage:
+      type: ephemeral
+    rack:
+      topologyKey: "failure-domain.beta.kubernetes.io/zone"
+    affinity:
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+          - matchExpressions:
+            - key: dedicated
+              operator: In
+              values:
+              - Kafka
+    logging:
+      type: inline
+      loggers:
+        log4j.rootLogger: INFO
+        #log4j.logger.kafka.authorizer.logger: INFO
+  zookeeper:
+    replicas: 3
+    readinessProbe:
+      initialDelaySeconds: 15
+      timeoutSeconds: 5
+    livenessProbe:
+      initialDelaySeconds: 15
+      timeoutSeconds: 5
+    storage:
+      type: ephemeral
+   logging:
+     type: inline
+     loggers:
+       zookeeper.root.logger: INFO
+  topicOperator:
+    logging:
+      type: inline
+      loggers:
+        rootLogger.level: INFO
+```
+
+Cluster Operator will deploy the StatfulSets for Zookeeper and Kafka as well as the Topic Operator.
+But only the Kafka pods will be scheduled to the dedicated nodes:
+
+```
+$ kubectl get pods -o wide
+NAME                                        READY     STATUS    RESTARTS   AGE       IP                NODE
+my-cluster-kafka-0                          2/2       Running   0          2m        192.168.158.65    ip-10-0-1-115.ec2.internal
+my-cluster-kafka-1                          2/2       Running   0          2m        192.168.221.65    ip-10-0-2-31.ec2.internal
+my-cluster-kafka-2                          2/2       Running   0          2m        192.168.226.67    ip-10-0-0-124.ec2.internal
+my-cluster-topic-operator-fb6cb47d-qqjrz    2/2       Running   0          1m        192.168.180.194   ip-10-0-1-18.ec2.internal
+my-cluster-zookeeper-0                      2/2       Running   0          2m        192.168.180.193   ip-10-0-1-18.ec2.internal
+my-cluster-zookeeper-1                      2/2       Running   0          2m        192.168.17.65     ip-10-0-0-236.ec2.internal
+my-cluster-zookeeper-2                      2/2       Running   0          2m        192.168.29.2      ip-10-0-2-61.ec2.internal
+strimzi-cluster-operator-586d499cd7-bzqsg   1/1       Running   0          9m        192.168.29.1      ip-10-0-2-61.ec2.internal
+```
```

Comment:
I think you need a summary paragraph to reiterate what you've explained (following the technicque often used in teaching where you say what you're going to explain, you explain it, and then you say what you explained).

In this blog we've explained that the noisy neighbour problem is and how Kubernetes/OpenShift's pod affinity features can be used to mitigate its effects. Pod affinity is not enough to eliminate noisy neighbours. To do that you have to use taints and tolerations to create a dedicated set of nodes for running your workload. Since Strimzi 0.5.0 it's possible to use these features with your Kafka and Kafka Connect clusters to get the best performance from the hardware available in your Kubernetes or OpenShift cluster.

Reply:
Yeah ... good point.